Hazel/src/Hazel/Core/Math/degrees.cpp

```diff
@@ -0,0 +1,72 @@
+#include "hzpch.h"
+
+
+
+#include "Hazel/Core/Base.h"
+#include "Hazel/Core/Math/degrees.h"
+#include "Hazel/Core/Math/radians.h"
+
+
+
+namespace Hazel
+{
+	Degrees::Degrees( const Radians& radians ) noexcept
+		:
+		m_Angle( radiansToDegrees( static_cast<float>( radians ) ) )
+	{}
+
+
+
+	[[nodiscard]]
+	constexpr Degrees::operator Radians( void ) const noexcept
+	{
+		return Radians( degreesToRadians( m_Angle ) );
+	}
+
+
+
+	[[nodiscard]]
+	constexpr Degrees Degrees::reduced( void ) const noexcept
+	{
+		auto angle = m_Angle;
+		while( angle > MAX_ANGLE )
+		{
+			angle -= ONE_FULL_ROTATION;
+		}
+		while( angle < MIN_ANGLE )
+		{
+			angle += ONE_FULL_ROTATION;
+		}
+		return Degrees( angle );
+	}
```

Comment:
This will probably go wrong for extremly large angles.

Reply:
Yeah, I forgot to take into account floating point precision for large values, it's easy just check for large values and do a straight division followed by multiplication by the fractional part (since you can't just use modulus for fp), but at that point you're losing too much precision so the actual angle just can no longer be stored anyway.  I considered just having a float for the angle that is always normalized between MIN and MAX angle and a separate integer for the number of full rotations and I also considered just reducing automatically after every math op. but bot approaches didn't seem ideal.  I can foresee many situations where a game might want to be able to determine that an object is rotated 1080 degrees rather than just storing 0 degrees for that.  I really wanted to just have integer degrees and not worry about a barely noticeable fractional angles.  Thinking about it now, the best thing would probably be to test for a threshold that uses the looped variation for smaller values where it will be faster than division and just divide whenever that would be faster.